### ISSUE TITLE
Access to price tier info once artwork has been submitted for approval

### ISSUE THREAD

**Issue by ZiggyAtl** [ocupop/comission-application#116]
When I'm on the individual artwork page in the Manage Artwork section (see screen shot below), it doesn't show which tiles are allocated to which price tier, so if someone wanted to check this, or even request a change, they can't see what they need to. I realize they can select the 'Adjust Pricing' button and it will then show that. However that's only before it's been submitted for approval. Once it's been submitted, the user can't access that button / page any more.

1) If this page showed the same image that shows on the Custom Pricing Tier page (ie with the various tier colors showing on the artwork), that would resolve that.

2) Once an artwork went live and had pieces sold, we would also want the artwork that showed in the individual artwork page to show which tiles were sold or unavailable

![Screen Shot 2022-10-31 at 12 14 13 PM](https://user-images.githubusercontent.com/112654923/199058805-c5ec2a85-c014-4c26-ace3-9981ec1c6acd.png)

**Comment by paulstroot** [comment]
Yes, got it! I will be working on a bit of a refactor on how this multi-step art creation process is handled, so this update may take a little while, but I'm on it!

**Comment by ZiggyAtl** [comment]
@paulstroot Unable to check this as I wasn't able to select any add any price tier information on the two sample pieces in the admin system.

**Comment by ZiggyAtl** [comment]
@paulstroot something doesn't seem to be working properly with this. I went back in to Mountain Landscape (which is still at In Progress status) and despite it having 100 tiles, there was only 1 selected at the higher tier, however the summary section said that there were -16 in stock of tier 1 and 66 in stock of tier 2. See screen shot
![Screen Shot 2022-11-06 at 5 31 17 PM](https://user-images.githubusercontent.com/112654923/200199232-005ade6e-ca9a-4edd-8878-a398f8e18bf9.png)

I was unable to check whether the system showed me the painting with all the tier pricing info showing, as the one which is at Awaiting CoMissioning stage doesn't have any price info selected other than tier 1. If you are able to drop a couple more of the previous artworks back on for me to check that would help.

**Comment by paulstroot** [comment]
I have not noticed this bug happening lately. There have been a lot of bug fixes and improvements that have hopefully fixed this. Keeping this issue open for a while in case you see it happening again.

If you do notice an inventory discrepancy, please make note of whether it was a newly created item, or if you had been actively editing pricing tiers and how.

**Comment by ZiggyAtl** [comment]
@paulstroot agreed I've not seen it recently, but I'll keep checking to make sure

**Comment by ZiggyAtl** [comment]
@paulstroot just spotted another one in an artwork i added in the last hour (Reach). Should be 3471 tiles (89x39) but it says inventory is over 19,000
![Screen Shot 2022-11-10 at 11 54 49 AM](https://user-images.githubusercontent.com/112654923/201158228-502d13c9-69b8-42a2-b602-1b9e2c7fd95e.png)

**Comment by ZiggyAtl** [comment]
@paulstroot and another. Just uploaded The Scream and its 120 x 120 which should be 14,400 tiles. But total inventory is 10,676. Is this something to do with some maybe listed as unavailable?

**Comment by ZiggyAtl** [comment]
@paulstroot  and another example. In 'Test 1' that had been submitted for Comissioning I added a piece of art (as there was none) and entered the rows and columns in shopify (89x39). I was unable to allocate any price tier data to any of the tiles, as I can't figure out how to do that. So everything should have been tier 1. However instead of 3471 inventory of tiles there are 3 in tier 1 and 1 in tier 2.

**Comment by paulstroot** [comment]
Thank you Bill. I am still looking into this

Have you been making any updates directly in Shopify? I'm not sure if I clearly stated this in our call, but there are some settings that should be made ONLY in the museum app, because changing them affects the values of a number of other settings. Have you by chance been updating the rows and columns for artwork directly in the Shopify admin instead of through the museum app?

Here is a list of things that should **not** be updated in Shopify:

- Rows
- columns
- pricing tiers (known as "Variants" in shopify)
- inventory
- Hi-res image path or Hi-Res dimensions
- Vendor Name

**Comment by ZiggyAtl** [comment]
@paulstroot thanks for the info and no you didn't mention that, so it's good to know. I was tinkering around with a few items in shopify, although I'm not sure I made row and column changes to the items we are referencing, but can't be sure on that.

**Comment by ZiggyAtl** [comment]
@paulstroot Found another example for you which I def know is not because it was altered in shopify. The Scream that you and I worked on in the training session. It should have 144,000 tiles, but only showing approx 10.5k

I asked a question in the training session and can't remember what the answer was. When it shows the dabs in stock in the app does that allow for any which have been classed as 'not available'?

**Comment by paulstroot** [comment]
Thanks. I believe I found the issue and fixed it. It was a tricky one to track down because it was only happening if you refractalized an image twice between page loads. Please keep an eye on this one and let me know if you see it pop up again.

**Comment by ZiggyAtl** [comment]
@paulstroot a bit more info about the above issue. Although the artwork lost all its price tier information, it at least maintained the number of DABs and therefore correct inventory. However I just realized Reach lost all its DAB / inventory info when it was submitted for comissioning. I tried it again (Reach 2) and this time I really upped the number of DABs (200 x 200). It did the same, and is just appearing as an inventory of 1 in shopify and in the app. The image in Shopify has no DAB outline info on, however the image in the app has at least retained the DAB borders on the image, even though it thinks it has a total inventory of 1 DAB

**Comment by paulstroot** [comment]
I took a deeper look into saving variants and discovered two bugs that I fixed. 

1) Old variants were not being successfully deleted, This should be fixed now. 

2) If two pricing tiers have the same name (i.e. "pricing tier 2"), there was an error saving them, but not an error with updating the inventory on the other pricing tiers, causing the numbers to be off. This could happen with the correct combination of creating and deleting pricing tiers as you make edits. I have added some form validation so that if two pricing tiers have the same name, the user sees a big red warning prompting them to re-name of of them.

Given these two updates, please review and let me know if you still see issues with updating inventory.
